### ISSUE TITLE
zergpool ccallium garlic prob

### ISSUE THREAD

**Issue by astitsa** [RainbowMiner/RainbowMiner#2169]
looks like rbm choose the wrong algo to mine

![c4](https://user-images.githubusercontent.com/97480060/192121886-bd717de9-8cfd-4680-b6f1-317c2c6394d1.PNG)
![c1](https://user-images.githubusercontent.com/97480060/192121887-dba06469-f5f0-4cfa-80aa-4b42f59fe8f9.PNG)
![c2](https://user-images.githubusercontent.com/97480060/192121888-09c83a0e-8a75-45e0-bd60-d9d822850353.PNG)
![c3](https://user-images.githubusercontent.com/97480060/192121889-894a7488-b646-45f8-81b5-1af499bd71bf.PNG)

**Comment by astitsa** [comment]
same problem with zpool ...

**Comment by astitsa** [comment]
![c2](https://user-images.githubusercontent.com/97480060/192122110-a83148dd-1ef5-4acd-bec3-84a0e0192021.PNG)
![c1](https://user-images.githubusercontent.com/97480060/192122112-991bded5-197b-41eb-911f-7b8f9cf83125.PNG)

**Comment by RainbowMiner** [comment]
How strange! Could you please upload a Debug file?

**Comment by astitsa** [comment]
[debug_2022-09-25.zip](https://github.com/RainbowMiner/RainbowMiner/files/9640177/debug_2022-09-25.zip)

**Comment by RainbowMiner** [comment]
(Related to/Same issue as #2169)

For a possible fix, please look here: https://github.com/RainbowMiner/RainbowMiner/issues/2170#issuecomment-1257157043

**Comment by astitsa** [comment]
ok thanks again i will keep you posted so you know if its fixed

**Comment by astitsa** [comment]
Ok it happened again but this time it is using my wallet address.

![image](https://user-images.githubusercontent.com/97480060/192577533-317367c3-9236-4cb6-b06b-1c2f8f083e97.png)
![image](https://user-images.githubusercontent.com/97480060/192577715-c9719f5f-7ad8-49ce-941a-db68fb3f90a2.png)


here is the debug file

[debug_2022-09-27.zip](https://github.com/RainbowMiner/RainbowMiner/files/9657805/debug_2022-09-27.zip)

**Comment by RainbowMiner** [comment]
How very weird! I can only imagine, that there is some overshooting profit value, that could cause a switch. I'll track the ZergPool Allium for some time, just to be sure.

**Comment by astitsa** [comment]
ccminer-x64 with allium (garlic) algorithm on zergpool

do you think they are doing this on purpose cause they know people use it for automation ?

**Comment by RainbowMiner** [comment]
> do you think they are doing this on purpose cause they know people use it for automation ?

I don't think so. It's most probably solo-miners that cause a sudden increase in the daily estimates.

**Comment by astitsa** [comment]
what i don't get is why the profit estimate is right on the algo list between each devices ?

you probably add couple variable like when the block was mined etc ?

**Comment by astitsa** [comment]
here is another situation with another pool (zpool) :

![image](https://user-images.githubusercontent.com/97480060/193093264-53b6e328-a3d0-4b0b-867b-9ab7458be99c.png)

![image](https://user-images.githubusercontent.com/97480060/193093438-9495c9f5-718b-4d7b-b4d0-db324ee553fd.png)

**Comment by astitsa** [comment]
bump

**Comment by astitsa** [comment]
i excluded zergpool, miningpoolhub and zpool but still mining that garlic shit O.O

![image](https://user-images.githubusercontent.com/97480060/194437000-b48921da-d18f-4132-ba26-a9f2dc55a1ba.png)
![image](https://user-images.githubusercontent.com/97480060/194437029-7bf8cda3-7d75-4127-adfd-d8ff6b800230.png)
[debug_2022-10-06.zip](https://github.com/RainbowMiner/RainbowMiner/files/9729230/debug_2022-10-06.zip)

**Comment by RainbowMiner** [comment]
Wtf! That's so weird. I'll put everything aside until this is solved.

**Comment by RainbowMiner** [comment]
Could you please upload the CcminerAllium stat files? `.\Stats\Miners\NVIDIA-CcminerAllium-*_Allium_HashRate.txt`

Also, in the meantime just put "Allium" on the "ExcludeAlgorithm" list in config.txt. Let's see, if that ends it or if another algorithm suddenly pops up.

**Comment by RainbowMiner** [comment]
To do deep analysis, I will need some more:
- please open the following URLs either on `localhost:4000` or `<rigip>:4000`
- save the results to json-files (right-click, then "save as..")
- send me the files by eMail at rbminer2018@gmail.com - _do not upload them here_, since the files will contain your wallets etc.

http://localhost:4000/miners
http://localhost:4000/activeminers
http://localhost:4000/fastestminers
http://localhost:4000/minersneedingbenchmark
http://localhost:4000/allpools
http://localhost:4000/pools

**Comment by astitsa** [comment]
I already tried to exclude allium but it mines something else on that party pool (zergpool, zpool, miningpoolhub). that's why i just exclude the 3 pool for now.

FYI i cleared data with all rbm bat file before running bench for this rig

Here are the stats file. I will send you the config file without wallet when i come back home.

[NVIDIA-CcminerAllium-GPU#00-GPU#02_Allium_HashRate.txt](https://github.com/RainbowMiner/RainbowMiner/files/9735800/NVIDIA-CcminerAllium-GPU.00-GPU.02_Allium_HashRate.txt)
[NVIDIA-CcminerAllium-GPU#00-GPU#02-GPU#08_Allium_HashRate.txt](https://github.com/RainbowMiner/RainbowMiner/files/9735802/NVIDIA-CcminerAllium-GPU.00-GPU.02-GPU.08_Allium_HashRate.txt)
[NVIDIA-CcminerAllium-GPU#01-GPU#03-GPU#04-GPU#05-GPU#06-GPU#07_Allium_HashRate.txt](https://github.com/RainbowMiner/RainbowMiner/files/9735803/NVIDIA-CcminerAllium-GPU.01-GPU.03-GPU.04-GPU.05-GPU.06-GPU.07_Allium_HashRate.txt)
[NVIDIA-CcminerAllium-GPU#08_Allium_HashRate.txt](https://github.com/RainbowMiner/RainbowMiner/files/9735805/NVIDIA-CcminerAllium-GPU.08_Allium_HashRate.txt)
[NVIDIA-CcminerAllium-GPU#00-GPU#01-GPU#02-GPU#03-GPU#04-GPU#05-GPU#06-GPU#07_Allium_HashRate.txt](https://github.com/RainbowMiner/RainbowMiner/files/9735806/NVIDIA-CcminerAllium-GPU.00-GPU.01-GPU.02-GPU.03-GPU.04-GPU.05-GPU.06-GPU.07_Allium_HashRate.txt)

**Comment by astitsa** [comment]
I think I've switched from combo MiningMode to device one without clearing all data. I found some data from combo mode ... It may have caused all this issue so I think I will try that before you go deep into this one.

I will keep you posted

**Comment by RainbowMiner** [comment]
This is a ZergPool algo-selection problem. ZergPool was removed from RainbowMiner on 2025-09-27 (b4f6f448), so it can no longer be reproduced. Closing as obsolete for the current master (v5.0.2.1) - please open a new issue if this still occurs. Thanks for the report!
